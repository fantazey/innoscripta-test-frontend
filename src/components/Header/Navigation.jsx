import React from "react";
import { NavLink } from "react-router-dom";

const Navigation = props => <nav style={{'backgroundColor': 'aliceblue'}}>
  <ul className={"nav  d-flex flex-row justify-content-between"}>
    {props.categories.map((item, index) =>
      (<li key={`nav_li_${index}`} className={"nav-item mx-3"}>
        <NavLink
          exact
          key={`navigation_${index}`}
          className='nav-link text-uppercase'
          activeClassName='active'
          to={item.path}>{item.label}
        </NavLink>
      </li>)
    )}
  </ul>
</nav>;

export default Navigation;