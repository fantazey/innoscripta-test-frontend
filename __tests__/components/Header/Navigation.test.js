import Navigation from "../../../src/components/Header/Navigation";

describe('Navigation', function () {
  let wrapper;
  let categories = [
    {
      path: '1',
      label: '1'
    },
    {
      path: '2',
      label: '2'
    }
  ];

  beforeEach(() => {
    wrapper = shallow(
      <Navigation categories={categories}/>
    )
  });

  it('should match snapshot', function() {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render all passed categories', function() {
    expect(wrapper.find('ul').children()).toHaveLength(categories.length);
  });

  it('should pass props correctly', function() {
    let child = wrapper.find('NavLink').first();
    expect(child.prop('to')).toEqual(categories[0].path);
    expect(child.text()).toEqual(categories[0].label);
  })
});
