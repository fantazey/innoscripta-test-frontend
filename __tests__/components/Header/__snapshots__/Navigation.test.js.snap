// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should match snapshot 1`] = `
<nav
  style={
    Object {
      "backgroundColor": "aliceblue",
    }
  }
>
  <ul
    className="nav  d-flex flex-row justify-content-between"
  >
    <li
      className="nav-item mx-3"
      key="nav_li_0"
    >
      <NavLink
        activeClassName="active"
        className="nav-link text-uppercase"
        exact={true}
        key="navigation_0"
        to="1"
      >
        1
      </NavLink>
    </li>
    <li
      className="nav-item mx-3"
      key="nav_li_1"
    >
      <NavLink
        activeClassName="active"
        className="nav-link text-uppercase"
        exact={true}
        key="navigation_1"
        to="2"
      >
        2
      </NavLink>
    </li>
  </ul>
</nav>
`;
